plothist/plotters.py

```diff
@@ -469,107 +321,237 @@ def plot_error_hist(hist, ax, **kwargs):
+    plot_hist(hist_2, ax=ax_main, label=x2_label, histtype="step")
+    ax_main.set_xlim(xlim)
+    ax_main.set_ylabel(ylabel)
+    ax_main.tick_params(axis="x", labelbottom="off")
+    ax_main.legend(framealpha=0.5)
+    _ = ax_main.xaxis.set_ticklabels([])
+    fig.subplots_adjust(hspace=0.125)
+
+    plot_comparison(
+        hist_1,
+        hist_2,
+        ax_comparison,
+        xlabel=xlabel,
+        x1_label=x1_label,
+        x2_label=x2_label,
+        comparison=comparison,
+        comparison_ylim=comparison_ylim,
+        ratio_uncertainty=ratio_uncertainty,
+    )
+
+    if save_as is not None:
+        fig.savefig(save_as, bbox_inches="tight")
+
+    return fig, ax_main, ax_comparison
+
+
+def plot_comparison(
+    hist_1,
+    hist_2,
+    ax,
+    xlabel="x1",
+    x1_label="x1",
+    x2_label="x2",
+    comparison="ratio",
+    comparison_ylim=None,
+    ratio_uncertainty="uncorrelated",
+):
+    """
+    Plot the comparison between two histograms.
+
+    Parameters
+    ----------
+    hist_1 : boost_histogram.Histogram
+        The first histogram for comparison.
+    hist_2 : boost_histogram.Histogram
+        The second histogram for comparison.
+    ax : matplotlib.axes.Axes
+        The axes to plot the comparison.
+    xlabel : str, optional
+        The label for the x-axis. Default is "x1".
+    x1_label : str, optional
+        The label for the first histogram. Default is "x1".
+    x2_label : str, optional
+        The label for the second histogram. Default is "x2".
+    comparison : str, optional
+        The type of comparison to plot ("ratio" or "pull"). Default is "ratio".
+    comparison_ylim : tuple or None, optional
+        The y-axis limits for the comparison plot. Default is None.
+    ratio_uncertainty : str, optional
+        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties)
+
+    Returns
+    -------
+    ax : matplotlib.axes.Axes
+        The axes with the plotted comparison.
+
+    See Also
+    --------
+    compare_two_hist : Compare two histograms and plot the comparison.
+
+    """
+    if not np.all(hist_1.axes[0].edges == hist_2.axes[0].edges):
+        raise ValueError("The bins of the compared histograms must be equal.")
 
     np.seterr(divide="ignore", invalid="ignore")
-    ratio = np.where(hist_1.values() != 0, hist_2.values() / hist_1.values(), np.nan)
-    ratio_variance = np.where(
-        hist_1.values() != 0,
-        hist_2.variances() / hist_1.values() ** 2
-        + hist_1.variances() * hist_2.values() ** 2 / hist_1.values() ** 4,
-        np.nan,
-    )
+    if comparison == "ratio":
+        ratio_values = np.where(
+            hist_2.values() != 0, hist_1.values() / hist_2.values(), np.nan
+        )
+        if ratio_uncertainty == "split" :
+            ratio_variance = (np.sqrt(hist_1.variances()) / hist_1.values()) ** 2
```

Comment:
Wrong. This is `h1_scaled_variance`, and we should divide by `hist_2.values()`, this is a bug in the current version of plothist. I understand that calling it `ratio_variance` helps later, but this is wrong name. The ratio variance is computed only in the `uncorrelated` case (and later in the `efficiency` case). We could define `h1_scaled_uncertainty` instead (not taking the square).